src/routes/layout-protected.route.tsx

```diff
@@ -0,0 +1,37 @@
+import { useAuth } from '#/modules/auth/hooks';
+import { Outlet, useNavigate } from 'react-router-dom';
+import { useAppDispatch } from '#/redux/hooks';
+import { createUser, removeUser } from '#/redux/auth/auth-slice';
+import CircularProgress from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+import { layoutProtectedStyles as styles } from '#/styles/layout-protected.styles';
+
+type Props = {
+  protectedRoute: boolean;
+};
+
+export function AuthLayout({ protectedRoute }: Props) {
+  const { data: user, isLoading, isError } = useAuth();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  if (isLoading) {
+    return (
+      <Box sx={styles.container}>
+        <CircularProgress sx={styles.spinner} />
+      </Box>
+    );
+  }
+
+  if (isError || !user) {
+    dispatch(removeUser());
```

Comment:
if the user logs out and they are on a public route, they will still remain on that public route, right? what do you think about redirecting them to the homepage?

Reply:
I think they should stay on the page they are on. it's quite disorienting to be redirected to homepage without any explanation just because there were problems with network or sth. but specifically for logging out, yes, they should be redirected.